apps/comps/components/agent-profile/index.tsx

```diff
@@ -92,6 +94,34 @@ export default function AgentProfile({
     }, "");
   }, [sortState]);
 
+  // Fetch EigenAI badge statuses for this agent across all competitions
+  // Returns a map of competitionId -> { isBadgeActive, signaturesLast24h }
+  const { data: eigenBadgeStatusMap } = useQuery(
+    tanstackClient.eigenai.getBadgeStatusesForAgent.queryOptions({
+      input: { agentId: id },
+    }),
+  );
+
+  // Transform the map to the format expected by CompetitionTable
+  type EigenBadgeStatus = {
+    competitionId: string;
+    isActive: boolean;
+    signaturesLast24h: number;
+  };
+
+  const eigenBadgesByCompetition = React.useMemo(() => {
+    if (!eigenBadgeStatusMap) return {} as Record<string, EigenBadgeStatus>;
```

Comment:
[nit] could probably do:
```ts
const result: Record<string, EigenBadgeStatus> = {};
if (!eigenBadgeStatusMap) return result;
```

Reply:
Right on - adjusted in [e14387a](https://github.com/recallnet/js-recall/pull/1657/commits/e14387ad73122597e2daf7086cc5c6d59795c191)